### PR TITLE
Add certificate_map to compute_target_ssl_proxy

### PR DESCRIPTION
**Pull request by pawel-grz-krawczyk**
Add certificate_map option to compute_target_ssl_proxy in order to support new certificate-manager (with wildcard support).

Similar PR: https://github.com/GoogleCloudPlatform/magic-modules/pull/5991

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

```release-note:enhancement
compute: added `certificate_map` to `compute_target_ssl_proxy` resource
```

### CODE REVIEW

**Comment by pawel-grz-krawczyk** [comment]
I ran acceptance test: 

```
make testacc TEST=./google TESTARGS='-run=TestAccComputeTargetSslProxy_update'
```

I get an error though:
```
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google -v -run=TestAccComputeTargetSslProxy_update -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccComputeTargetSslProxy_update
    provider_test.go:445: One of GOOGLE_CREDENTIALS, GOOGLE_CLOUD_KEYFILE_JSON, GCLOUD_KEYFILE_JSON, GOOGLE_APPLICATION_CREDENTIALS, GOOGLE_USE_DEFAULT_CREDENTIALS must be set for acceptance tests
--- FAIL: TestAccComputeTargetSslProxy_update (0.00s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-google/google   0.312s
FAIL
make: *** [GNUmakefile:15: testacc] Error 1
```

Any ideas how to fix it?

File: mmv1/third_party/terraform/tests/resource_compute_target_ssl_proxy_test.go.erb [hunk 1]
```diff
@@ -25,22 +35,34 @@ func TestAccComputeTargetSslProxy_update(t *testing.T) {
...
-					testAccCheckComputeTargetSslProxy(
-						t, "google_compute_target_ssl_proxy.foobar", "PROXY_V1", cert2),
+					testAccCheckComputeTargetSslProxyExists(
+						t, "google_compute_target_ssl_proxy.foobar", &proxy),
+					testAccCheckComputeTargetSslProxyHeader(t, "PROXY_V1", &proxy),
+					testAccCheckComputeTargetSslProxyHasSslCertificate(t, cert2, &proxy),
+				),
+			},
+			{
+				Config: testAccComputeTargetSslProxy_certificateMap(resourceSuffix),
```
Comment (c2thorn) [inline mmv1/third_party/terraform/tests/resource_compute_target_ssl_proxy_test.go.erb:54]: Can we test the update function of `certificate_map` similar to how the `ssl_certificates` field is tested?

**Review by c2thorn — CHANGES_REQUESTED** [review]
Thanks for the contribution @pawel-grz-krawczyk. Just requesting an update to the test

>     provider_test.go:445: One of GOOGLE_CREDENTIALS, GOOGLE_CLOUD_KEYFILE_JSON, GCLOUD_KEYFILE_JSON, GOOGLE_APPLICATION_CREDENTIALS, GOOGLE_USE_DEFAULT_CREDENTIALS must be set for acceptance tests

You'll need to set some sort of credential, check out https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#running-tests

You are correct in assuming other failing tests are not related

File: mmv1/third_party/terraform/tests/resource_compute_target_ssl_proxy_test.go.erb (hunk 1 above)
Comment (pawel-grz-krawczyk) [reply mmv1/third_party/terraform/tests/resource_compute_target_ssl_proxy_test.go.erb:54]: Could you elaborate? What exactly does "similar" mean? 

The only difference that I see is that in `certificate_map` case we use different resource name format than in `ssl_certificates`:

```
certificate_map = "//certificatemanager.googleapis.com/${google_certificate_manager_certificate_map.map.id}"

vs

ssl_certificates = [google_compute_ssl_certificate.bar.name]
```

I'm not sure whether this format is necessary, I've seen it in one of tests though: `mmv1/third_party/terraform/tests/resource_compute_target_https_proxy_test.go.erb`

Comment (pawel-grz-krawczyk) [reply mmv1/third_party/terraform/tests/resource_compute_target_ssl_proxy_test.go.erb:54]: BTW: I no longer have these errors about missing credentials. I must have fixed while working on the task. `TestAccComputeTargetSslProxy_update` test passes.

Comment (c2thorn) [reply mmv1/third_party/terraform/tests/resource_compute_target_ssl_proxy_test.go.erb:54]: Sorry for the ambiguity. In `testAccComputeTargetSslProxy_basic1`, `ssl_certificates` is set to `google_compute_ssl_policy.foo.self_link`. It is updated to `google_compute_ssl_certificate.bar.name` in `testAccComputeTargetSslProxy_basic2`. 

I wanted to test out the update functionality of `certificate_map` (especially as it uses a `update_verb` and separate `update_url`) by adding a step where we update the value. However you think that should be done

Comment (pawel-grz-krawczyk) [reply mmv1/third_party/terraform/tests/resource_compute_target_ssl_proxy_test.go.erb]: Done (similar to `testAccComputeTargetSslProxy_basic1` and `testAccComputeTargetSslProxy_basic2`)

**Review by c2thorn — APPROVED** [review]
Thank you!

**Pull request merged**